src/components/ScenarioParameters/components/ScenarioParametersInputs/ScenarioSelect.js

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+import React, { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import PropTypes from 'prop-types';
+import { Grid } from '@mui/material';
+import { Select } from '@cosmotech/ui';
+import { useSortedScenarioList } from '../../../../hooks/ScenarioListHooks';
+import { useCurrentScenarioId } from '../../../../state/hooks/ScenarioHooks';
+import { TranslationUtils } from '../../../../utils';
+
+export const ScenarioSelect = ({ parameterData, context, parameterValue, setParameterValue, isDirty }) => {
+  const { t } = useTranslation();
+  const scenarioList = useSortedScenarioList();
+  const currentScenarioId = useCurrentScenarioId();
+  const runTemplateWhiteList = parameterData.options?.runTemplateFilter;
+
+  const filteredScenarioList = useMemo(() => {
+    if (runTemplateWhiteList == null || runTemplateWhiteList?.length === 0) return scenarioList;
+    const filteredScenarioTempList = [];
+    scenarioList.forEach((scenario) => {
+      const simulationRun = scenario?.lastRun?.csmSimulationRun;
+      if (
+        runTemplateWhiteList.includes(scenario.runTemplateId) &&
+        simulationRun != null &&
+        scenario.id !== currentScenarioId
+      )
+        filteredScenarioTempList.push({ key: simulationRun, label: scenario.name });
+    });
+    return filteredScenarioTempList;
```

Comment:
It's a bit confusing : if there is no `runTemplateWhiteList`, we return the whole `scenarioList` but in the second condition `runTemplateWhiteList` is mandatory. Also, in the first case we return a scenarios object with all properties and in the second, a `key/label` pair (the shape the `Select` component waits) I don't see how it works... :thinking: 

Finally, is it possible to refactor the function with `filter()` and `map()` methods to be more consice ? :pray: